src/mpid/ch4/netmod/ucx/ucx_init.c

```diff
@@ -358,16 +368,20 @@ int MPIDI_UCX_post_init(void)
 {
     int mpi_errno = MPI_SUCCESS;
 
-    for (int i = 1; i < MPIDI_UCX_global.num_vnis; i++) {
-        mpi_errno = init_worker(i);
+    if (MPIDI_UCX_global.num_vnis > 1) {
```

Comment:
I don't see how this change is related to the rest of the commit.

Reply:
I totally don't recall how this change was added. The main seems works fine without it. It makes sense though as it at least make the code cleaner. Let me split this commit.